R/dataset.R

```diff
@@ -88,6 +88,77 @@ setMethod("notes<-", "CrunchDataset", function (x, value) {
     invisible(setEntitySlot(x, "notes", value))
 })
 
+#' Get and set the market size for Crunch datasets
+#'
+#' Crunch Datasets allow you to set a target population size in order to extrapolate
+#' population estimates from survey percentages. This function lets you work with
+#' the population size and magnitude.
+#'
+#' @param x a Crunch Dataset
+#' @param value For the setters, the `size` or `magnitude` to be set
+#' @param size the target population size, to remove a population set to `NULL`
+#' @param magnitude the order of magnitude with which to display the population
+#' size. Must be either `3`, `6`, or `9` for thousands, millions, and billions respectively.
+#' @return `popSize` and `popMagnitude` return the population size or
+#' magnitude. `setPopultion` returns the modified dataset.
+#' @name population
+#' @aliases popSize popMagnitude setPopulation popSize<- popMagnitude<-
+NULL
+
+#' @rdname population
+#' @export
+setMethod("popSize", "CrunchDataset", function (x) {
+    size <- settings(x)$population$size
+    return(ifelse(size == 0, NULL, size)) #the API value 0 stands for "No Population set"
+})
+
+#' @rdname population
+#' @export
+
+setMethod("popSize<-", "CrunchDataset", function (x, value) {
+    setPopulation(x, size = value)
+})
+
+#' @rdname population
+#' @export
+setMethod("popMagnitude", "CrunchDataset", function (x) {
+    return(settings(x)$population$magnitude)
+})
+
+#' @rdname population
+#' @export
+setMethod("popMagnitude<-", "CrunchDataset", function (x, value) {
+   setPopulation(x, magnitude = value)
+})
+
+#' @rdname population
+#' @export
+setMethod("setPopulation", "CrunchDataset", function (x, size, magnitude) {
+    pop <- settings(x)$population
+    if (missing(magnitude)) {
+        if (is.null(pop$magnitude)) {
+            warning("Dataset magnitude not set, defaulting to thousands")
+            magnitude <- 3
+        } else {
+            magnitude <- pop$magnitude
+        }
+    }
+    if (missing(size)) {
+        if (is.null(pop$size)) {
+            stop("Dataset does not have a population, please set one before attempting to change magnitude")
```

Comment:
use `halt`

Reply:
Why do we care that someone is setting magnitude before setting size? Does something break?

Reply:
It's really that you need to supply both size and magnitude together and there's not a sensible way to supply a default population. So if there's no magnitude we set it to thousands and warn the user, but it there's no population we just error.